### DISCUSSION TITLE
Node Classification of N Graphs

### DISCUSSION THREAD

**Discussion by elbasir** [pyg-team/pytorch_geometric#6391] [General]
Hello,

I know GCN can predict nodes in a graph but can you consider the following scenario that I have:

1- Assume there are N patients, where each patient has his/her own PPI network and I want to classify nodes "Proteins that are mutated in each patient"  in each PPI network. Can GCN handle this by passing multiple graphs (Multiple PPI networks)? 

2- If the answer is Yes, how to predict nodes for a new patient in the test phase as each PPI network is unique per patient in terms of mutated protein nodes VS non-mutated protein nodes?

Thanks!

**Comment by rusty1s** [comment]
1. Each patient does have a graph for which we do node classification, while we can also batch multiple graphs together. This is very similar to the PPI example we have in `examples/ppi.py`.
2. You would just use the learned GNN and predict nodes in the new unseen network.

**Comment by elbasir** [comment]
Thanks for your answers. I still have a couple of questions.

1- How can I validate the results per patient in the training stage? 
2- How the shape of the prediction output looks like to tell about each patient's classified nodes network?

**Reply by LukeLIN-web** [reply]
1. It seems strange, why you validate the results at the training set? 
2.  You can print the shape `print(out.shape)`

**Reply by elbasir** [reply]
I am still not clear how GCN can distinguish between every validation & test split per graph if I pass multiple graphs together.

**Reply by rusty1s** [reply]
Usually, people run a round of validation after each training epoch. This is also what the `examples/ppi.py` script is doing.

**Reply by elbasir** [reply]
Clear, thanks a lot.
